src/server/server_family.cc

```diff
@@ -1946,20 +2020,34 @@ void ServerFamily::ReplicaOf(CmdArgList args, ConnectionContext* cntx) {
     return;
   }
 
-  // Flushing all the data after we marked this instance as replica.
-  Transaction* transaction = cntx->transaction;
-  transaction->Schedule();
+  if (flush_db == ShouldFlushDb::kFlush) {
+    // Flushing all the data after we marked this instance as replica.
+    Transaction* transaction = cntx->transaction;
+    transaction->Schedule();
 
-  auto cb = [](Transaction* t, EngineShard* shard) {
-    shard->db_slice().FlushDb(DbSlice::kDbAll);
-    return OpStatus::OK;
-  };
-  transaction->Execute(std::move(cb), true);
+    auto cb = [](Transaction* t, EngineShard* shard) {
+      shard->db_slice().FlushDb(DbSlice::kDbAll);
+      return OpStatus::OK;
+    };
+    transaction->Execute(std::move(cb), true);
+  }
 
   // Replica sends response in either case. No need to send response in this function.
   // It's a bit confusing but simpler.
   lk.unlock();
-  error_code ec = new_replica->Start(cntx);
+  error_code ec{};
+
+  switch (on_err) {
+    case ActionOnConnectionFail::kReturnOnError:
+      ec = new_replica->Start(cntx);
+      break;
+    case ActionOnConnectionFail::kContinueReplication:  // set DF to replicate, and forget about it
+      ec = new_replica->EnableReplication(cntx);
+      break;
+    default:
+      CHECK(false) << "Uncovered case of ActionOnConnectionFail: " << on_err;
```

Comment:
Cool fact: if you remove the `default` clause, we'll actually get a compile warning in the case we don't check all possible `enum` cases, so it's to remove it